code/game/objects/items/storage/book.dm

```diff
@@ -207,6 +207,8 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "burning",
 
 	else if(istype(A, /obj/item/soulstone) && !iscultist(user))
 		var/obj/item/soulstone/SS = A
+		if(SS.purified)
+			return
 		to_chat(user, "<span class='notice'>You begin to exorcise [SS].</span>")
 		playsound(src,'sound/hallucinations/veryfar_noise.ogg',40,1)
 		if(do_after(user, 40, target = SS))
```

Comment:
```
if(do_after(user, 40, target = SS))
	if(SS.purified)
		return
```

Reply:
I did this and it didn’t work

Reply:
It will still stack here but after it purifies the rest return.

If you're trying to avoid stacking at all you'll need something like an inuse var that goes to true BEFORE the do_after.

Reply:
these are still all incorrect, do after takes a call back that is checked once each tick, you can immediately check and continuously check the status of SS by passing in a call back that checks SS purified